gitbootcamp.rst

```diff
@@ -0,0 +1,244 @@
+
+   $ git clone git@github.com:UserName/cpython.git
+   $ cd cpython
+   $ git remote add upstream git@github.com:python/cpython.git
+
+
+Listing the Remote Repositories
+-------------------------------
+
+   $ git remote -v
+
+
+Creating and Switching Branches
+-------------------------------
+
+.. note::
+   Never commit directly to the `master` branch.
+
+Create a new branch::
+
+   $ git checkout -b some-branch master # creates a new branch off master
+
+which is equal to::
+
+   $ git branch some-branch # create 'some-branch', without checking it out
+   $ git checkout some-branch # check out 'some-branch'
+
+To find out which branch you are in now::
+
+   $ git branch
+
+The current branch will have an asterisk next to the branch name.  Note, this
+will list all of your local branches.
+
+To list all the branches, including the remote branches::
+
+   $ git branch -a
+
+To switch to a different branch::
+
+   $ git checkout another-branch-name
+
+
+Delete Local Branch
+-------------------
+
+To delete branch that you no longer need::
+
+   $ git branch -D branch-to-delete
+
+
+Staging and Committing Files
+----------------------------
+
+1. To show the current changes::
+
+      $ git status
+
+2. To stage the files to be included in your commit::
+
+      $ git add /path/to/file1 path/to/file2 path/to/file3
+
+3. To commit the files that have been staged (done in step 2)::
+
+      $ git commit -m "This is the commit message. Prefix it with bpo-XXXX"
+
+
+Reverting Changes
+-----------------
+
+To revert changes to a file that has not been committed yet::
+
+    $ git checkout path/to/file
+
+
+If the change has been committed, and now you want to reset it to whatever
+the origin is at::
+
+   $ git reset --hard HEAD
+
+
+Stashing Changes
+----------------
+
+To stash away changes that are not ready to be committed yet::
+
+   $ git stash
+
+
+To re-apply last stashed change::
+
+   $ git stash pop
+
+
+Pushing Changes
+---------------
+
+Once your changes are ready for a review or a pull request, you'll need to push
+them to the remote repository.
+
+::
+
+   $ git checkout some-branch
+   $ git push origin some-branch
+
+
+Creating a Pull Request
+-----------------------
+
+1. Go to https://github.com/python/cpython.
+
+2. Click ``compare across forks`` link.
+
+3. Select the base fork: ``python/cpython`` and base branch: ``master``
+
+4. Select the head fork: ``UserName/cpython`` and base branch: the branch
+   containing your changes.
+
+5. Press ``Create Pull Request`` button.
+
+
+Syncing With Upstream
+---------------------
+
+Scenario:
+
+- You forked cpython repository some time ago.
+- Time passes.
+- There have been new commits made in upstream cpython repository.
+- Your forked cpython repository is no longer up to date.
+- You now want to update your forked cpython repository to be the same as
+  upstream.
+
+Solution::
+
+   $ git checkout master
+   $ git fetch upstream
+   $ git rebase upstream/master
```

Comment:
I think @pfmoore picked up on my initial confusion. As the instructions are just to keep master in sync then the rebase step seemed odd to me since you're not rebasing any previous commits you made locally. Maybe a comment at the start saying "there are alternative ways to perform the synchronization, but the approach taken to provide steps that are the same across scenarios"?